### DISCUSSION TITLE
Alist前端如何部署到CloudFlare Pages上？

### DISCUSSION THREAD

**Discussion by Simonzhang8** [AlistGo/alist#6680] [Q&A] [answered]
我看alist官方的demo地址是alist.pages.dev，说明安装在了Pages上面。但是alist不是必须部署在服务器上吗，这个是怎么做到部署到CloudFlare Pages上的？

**Comment by ix520** [comment]
我也有这种想法，刚才看了一下alist.pages.dev网页的源码，看到了 api:"https://al.nn.ci/"，alist.pages.dev与https://al.nn.ci/内容一致，我想应该是通过api来转发到alist.pages.dev上面，放到本地运行了一下正常，然后我把 api:"https://al.nn.ci/"中的链接换成了其他alist的链接，标题加载的是改动后alist链接的标题，但路径是storage
/
emulated
/
0
/
Android
/
data
/
top.webcat.editor
/
files
/
WebCat
/
project
/
111111111122131312122121212
/
alist2.html也就是我运行爬取的alist.pages.dev的html运行路径，因为不懂也没具体测试，希望能提供帮助，有结果了踢我一下👀👀👀

**Reply by Simonzhang8** [reply]
想问下您这个html文件是哪里获取的？我之前是直接把alist.pages.dev的html原码从浏览器上拷贝下来，api改成自己的，但是运行不起来。

**Reply by Simonzhang8** [reply]
我找到了办法。直接把alist-web这个项目fork一下，把里面index.html里的api填入自己的alist网址，然后在cloudflare pages里运行npm run build就可以了。不过现在有两个问题，一个是编译出来的是英文版的没有中文，还有一个是@manage设置里的所有个性化设置比如head，body，logo这些都无效了

**Reply by Simonzhang8** [reply]
head和body好像得在index.html里面编辑了，logo和语言暂时还没弄

**Reply by Simonzhang8** [reply]
我刚才成功的Pages的网站是alist-zxm.pages.dev，你可以参考下

**Reply by ix520** [reply]
好的，感谢

**Comment by Simonzhang8** [comment, marked as answer]
> 现在已经修改标题了，大家看清楚是Alist的前端部署到Cloudflare Pages，实现的效果是前后端的分离，后端暂时还是需要依赖服务器的。原来的标题说的不是很清楚，造成部分朋友误解了认为Alist可以不依赖服务器直接在Pages上使用。

我找到了办法。直接把alist-web这个项目fork一下，把里面index.html里的api填入自己的alist网址
**_注意是Alist原本的网址（服务器反代后的地址，或者服务器的ip地址但cf要求80或443端口），不是Pages的地址！！！换句话说Cloudflare Pages只是给Alist套壳了一个前端，本身不能独立运行（如果有大佬有办法不依赖服务器运行后端的话欢迎交流）_** 
然后在cloudflare pages里运行npm run build就可以了。不过现在有两个问题，一个是编译出来的是英文版的没有中文，还有一个是@manage设置里的所有个性化设置比如head，body，logo这些都无效了

> 这个比较适合有分开部署前后端，或者不想手动编译前后端但又想自定义前端，或者希望将更多静态资源托管到cdn等需求的用户。同理，类似的方法也可以部署到Github Pages（目前测试有些问题，可能需要大神帮助）、Netlify、Vercel或虚拟主机等平台上（Netlify和Vercel与CF Pages一样也可以自动编译，这里用了CF Pages是因为Cloudflare无流量限额，Vercel刚才试了也行（注意需要多一句pnpm install），大家可以根据自己喜好选择托管平台；虚拟主机不能自动编译，需要本地编译完后扔到平台上，不推荐）

**Reply by Simonzhang8** [reply]
head和body好像得在index.html里面编辑了，logo和语言暂时还没弄

**Reply by anwen-anyi** [reply]
```JavaScript
    <script>
      window.ALIST = {
        cdn: undefined,
        monaco_cdn: undefined,
        base_path: undefined,
        api: undefined,
        main_color: undefined,
      }
      window.__dynamic_base__ = window.ALIST.cdn || ""
    </script>
```
` api: undefined,` 这个吗

**Reply by ix520** [reply]
是的

**Reply by Simonzhang8** [reply]
> ```js
>     <script>
>       window.ALIST = {
>         cdn: undefined,
>         monaco_cdn: undefined,
>         base_path: undefined,
>         api: undefined,
>         main_color: undefined,
>       }
>       window.__dynamic_base__ = window.ALIST.cdn || ""
>     </script>
> ```
> 
> ` api: undefined,` 这个吗

对的，undefined改成alist服务器的地址，地址外面需要套个单引号，可以参考alist.pages.dev的html源码里的写法（但是目前测试下来，好像也只能80和443端口，估计跟alist代理那个一样是cf的限制）

**Comment by anwen-anyi** [comment]
翻译在线搞倒是不会，本地我倒是可以，倒是可以从本地上传到`pages` 嘿嘿

**Reply by Simonzhang8** [reply]
对的，本地全部弄好再一起传到cf pages上也可以（包括github pages，netlify等等平台应该都可以）。不过我个人是想着既然在线让cf执行编译就一起编译掉最好，这样的好处是github上源码一编辑，cf能自动重新编译发布，倒是蛮省时省力的。不过cf那个编译好像只能填一个语句进去，所以有点犯难

**Reply by anwen-anyi** [reply]
> 对的，本地全部弄好再一起传到cf pages上也可以（包括github pages，netlify等等平台应该都可以）。不过我个人是想着既然在线让cf执行编译就一起编译掉最好，这样的好处是github上源码一编辑，cf能自动重新编译发布，倒是蛮省时省力的。不过cf那个编译好像只能填一个语句进去，所以有点犯难

第一次用这个Pages 在哪儿输入编译命令呀…方便截个图吗😘

**Reply by Simonzhang8** [reply]
> 第一次用这个Pages 在哪儿输入编译命令呀…方便截个图吗😘
![image](https://github.com/alist-org/alist/assets/165749751/b4d1115a-7fbd-47af-8716-8df7bfab39cb)

**Reply by Simonzhang8** [reply]
创建的时候要选择导入现有 Git 存储库，从Github/Gitlab中导入；如果从本地上传的话好像不能在线构建

**Reply by anwen-anyi** [reply]
> 创建的时候要选择导入现有 Git 存储库，从Github/Gitlab中导入；如果从本地上传的话好像不能在线构建

好的~ 先放起来吧 过两天回家了再看看

**Comment by xingnengmao666** [comment]
alist-web-sqnb.pages.dev 这是我的网站，但是在配置完成后进入网站，提示405错误，且没有admin的权限

**Reply by Simonzhang8** [reply]
> alist-web-sqnb.pages.dev 这是我的网站，但是在配置完成后进入网站，提示405错误，且没有admin的权限

api不是填Cloudflare Pages的地址，而是你原本alist部署的网址：比如alist官方demo的alist.pages.dev对应的原部署地址是al.nn.ci那么api填的就是al.nn.ci而不是alist.pages.dev
![image](https://github.com/user-attachments/assets/cde62be4-5e26-4258-a761-950b4259bf49)

**Reply by xingnengmao666** [reply]
懂了，谢谢！

**Comment by zyjtest001git** [comment]
![image](https://github.com/user-attachments/assets/77c775ff-d9bd-4477-a2da-10e72d5c2baa)
请问一下，这种错误是什么做出了啊，这是我部署的地址alist-web-9l8.pages.dev

**Reply by Simonzhang8** [reply]
跟上面一样，api要填的是原本的alist地址而不是pages的地址
![image](https://github.com/user-attachments/assets/e3558fa4-e935-4b49-8441-e65bd58d5d19)

**Comment by tujiangh** [comment]
蹲一下，哪位大佬有不用服务器的方法踢一下，谢谢😊

**Comment by tujiangh** [comment]
或者有没有其他什么不用服务器就能部署的方法，推荐一下，谢谢

**Reply by ix520** [reply]
不用服务器还知不知道，没有服务器的话你可以去尝试注册一下serv00，一个免费10年的小服务器，网上有此服务器alist搭建教程

**Reply by Simonzhang8** [reply]
> 不用服务器还知不知道，没有服务器的话你可以去尝试注册一下serv00，一个免费10年的小服务器，网上有此服务器alist搭建教程

注册不了😢

**Reply by ix520** [reply]
给我一个联系方式吧，或者邮箱，我尝试帮你注册一下

**Comment by 122cygf** [comment]
无服务器的话，试试腾讯和阿里的云函数？可以把alist二进制文件传到对象存储，然后挂载到云函数上，启动脚本直接运行二进制文件
以上只是我的一个猜想，包括可用性、速度、产生的费用都未考量

**Reply by Simonzhang8** [reply]
听起来有点麻烦，感觉好像还不如服务器划算😂 不知道有没有动手党试试看

**Reply by 122cygf** [reply]
我已经尝试部署在阿里的函数计算了，速度还是很快的

**Reply by Simonzhang8** [reply]
> 我已经尝试部署在阿里的函数计算了，速度还是很快的

可以分享一下方法吗😘😘😘

**Reply by alan16742** [reply]
> 我已经尝试部署在阿里的函数计算了，速度还是很快的

话说你的加载速度快吗，正常来说冷启动不应该是一分多钟吗，你是修改了什么吗

**Reply by 122cygf** [reply]
> > 我已经尝试部署在阿里的函数计算了，速度还是很快的
> 
> 可以分享一下方法吗😘😘😘

用的就是我说的那个方法

**Reply by 122cygf** [reply]
> > 我已经尝试部署在阿里的函数计算了，速度还是很快的
> 
> 话说你的加载速度快吗，正常来说冷启动不应该是一分多钟吗，你是修改了什么吗

没修改什么，直接用二进制文件，冷启动也不慢

**Reply by alan16742** [reply]
你说的不慢是多久，我之前使用Onedrive测试给我卡了两分多钟，我服了这个速度

**Reply by 122cygf** [reply]
> 你说的不慢是多久，我之前使用Onedrive测试给我卡了两分多钟，我服了这个速度

我这里冷启动0.91毫秒，冷启动很快的，多出来的时间用来加载onedrive文件了吧

**Reply by alan16742** [reply]
你说是前端的加载时间吗，我指的是后端的加载时间，不会显示  ` 获取设置失败：请稍后，正在加载储存` 吗

**Reply by 122cygf** [reply]
> 你说是前端的加载时间吗，我指的是后端的加载时间，不会显示 ` 获取设置失败：请稍后，正在加载储存` 吗

没有

**Reply by alan16742** [reply]
你是自己写了一个 `sh start.sh` 来启动吗

**Comment by VOfficial404** [comment]
可以下载releases编译好的前端文件 这样就省去了build了

**Comment by tujiangh** [comment]
嗯，有空试试

yyyyt ***@***.***> 于2024年9月22日周日 09:02写道：
